google-cloud-spanner/src/main/java/com/google/cloud/spanner/SpannerExceptionFactory.java

```diff
@@ -183,10 +193,13 @@ private static SpannerException newSpannerExceptionPreformatted(
       case ABORTED:
         return new AbortedException(token, message, cause);
       case NOT_FOUND:
-        if (message != null && message.contains("Session not found")) {
-          return new SessionNotFoundException(token, message, cause);
-        } else if (message != null && DATABASE_NOT_FOUND_MSG_PATTERN.matcher(message).matches()) {
-          return new DatabaseNotFoundException(token, message, cause);
+        ResourceInfo resourceInfo = extractResourceInfo(cause);
+        if (resourceInfo != null) {
```

Comment:
what about Instance Not Found ?

Reply:
The client library does not currently handle `InstanceNotFound` specifically, so it is currently not recognized as a separate error. I'll open a separate issue for it, as it could cause the same problems as `DatabaseNotFound` if someone deletes an instance while the client library has an active connection with a database on that instance.